src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WrapWithTag/WrapWithTagEndpoint.cs

```diff
@@ -98,7 +102,36 @@ internal class WrapWithTagEndpoint : IWrapWithTagEndpoint
             var response = await _languageServer.SendRequestAsync(LanguageServerConstants.RazorWrapWithTagEndpoint, parameter).ConfigureAwait(false);
             var htmlResponse = await response.Returning<WrapWithTagResponse>(cancellationToken).ConfigureAwait(false);
 
+            if (htmlResponse.TextEdits is not null)
+            {
+                htmlResponse.TextEdits = await CleanUpTextEditsAsync(documentContext, htmlResponse.TextEdits, cancellationToken).ConfigureAwait(false);
+            }
+
             return htmlResponse;
         }
+
+        /// <summary>
+        /// Sometimes the Html language server will send back an edit that contains a tilde, because the generated
+        /// document we send them has lots of tildes. In those cases, we need to do some extra work to compute the
+        /// minimal text edits
```

Comment:
Do the minimal text edits we compute still contain the tilde? How do we ensure that those tilde's don't leak into the top level document? OR is the answer that we do a "best attempt"?

Reply:
> Do the minimal text edits we compute still contain the tilde?

Hopefully not, though I deliberately didn't filter if they did, so worst case with this PR is the same behaviour users get now.

> How do we ensure that those tilde's don't leak into the top level document? OR is the answer that we do a "best attempt"?

Yes, essentially we're just relying on the knowledge (also known as "hope") that our minimal text differ is actually minimal. If Web Tools used a minimal differ we wouldn't have this issue, but they're not actually doing anything wrong so forcing it on to their side seemed a bit egregious.